src/components/Attributes/Attribute.js

```diff
@@ -0,0 +1,267 @@
+                    validate={validators(
+                        {
+                            hasValue: required,
+                            positiveInteger: true,
+                            unique,
+                        },
+                        uniqueValidator
+                    )}
+                />
+            )
+        case 'INTEGER_NEGATIVE':
+            return (
+                <TextField
+                    required={required}
+                    name={name}
+                    label={label}
+                    initialValue={value}
+                    validate={validators(
+                        {
+                            hasValue: required,
+                            negativeInteger: true,
+                            unique,
+                        },
+                        uniqueValidator
+                    )}
+                />
+            )
+        case 'DATE':
+            return (
+                <DateField
+                    required={required}
+                    name={name}
+                    label={label}
+                    initialValue={value}
+                    validate={validators(
+                        {
+                            hasValue: required,
+                            unique,
+                        },
+                        uniqueValidator
+                    )}
+                />
+            )
+        case 'BOOLEAN':
+            // SingleSelectField throws an error if its value does not correspond to an option
+            return (
+                <SingleSelectField
+                    required={required}
+                    clearable={!required}
+                    name={name}
+                    label={label}
+                    options={[
+                        { label: i18n.t('Yes'), value: 'true' },
+                        { label: i18n.t('No'), value: 'false' },
+                    ]}
+                    initialValue={
+                        ['true', 'false'].includes(value) ? value : undefined
+                    }
+                    validate={validators({
+                        hasValue: required,
+                    })}
+                />
+            )
+        case 'TRUE_ONLY':
+            return (
+                <CheckboxField name={name} label={label} initialValue={value} />
+            )
+        // Use TEXT as fallback field type. This way all attributes will always
+        // be editable, albeit not necessarily enforcing the correct formatting
+        case 'TEXT':
+        default:
+            return (
+                <TextField
+                    required={required}
+                    name={name}
+                    label={label}
+                    initialValue={value}
+                    validate={validators(
+                        {
+                            hasValue: required,
+                            unique,
+                        },
+                        uniqueValidator
```

Comment:
I couldn't figure out a nicer way of passing the unique validator to `validators`